awx/api/serializers.py

```
@@ -2270,6 +2291,39 @@ class Meta:
 
 
+_VALID_PROXY_SCHEMES = ('http', 'https', 'socks', 'socks4', 'socks4a', 'socks5', 'socks5h')
+
+
+def _validate_proxy_url(value):
+    """Validate that *value* is a well-formed proxy URL (scheme + hostname)."""
+    parsed = urllib.parse.urlsplit(value)
+    if parsed.scheme.lower() not in _VALID_PROXY_SCHEMES or parsed.hostname is None:
+        raise serializers.ValidationError(_('Proxy must be an HTTP, HTTPS, or SOCKS URL.'))
+    try:
+        parsed.port
```

Ruff (0.15.10)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)